specification/eventgrid/Azure.Messaging.EventGrid/client.tsp

```diff
@@ -156,6 +156,7 @@ namespace Customizations {
 //Mark models as public (except ReceiveResult, PublishResult)
```

Comment:
Can we delete the mention of PublishResult? Also, can we add a comment saying that ReceiveResult ends up being public via customization in all languages, but that we leave it internal here in order to leverage the Azure.Core CloudEvent type (except for Java)?